core/src/main/java/org/elasticsearch/monitor/os/OsStats.java

```diff
@@ -166,22 +170,70 @@ public static Cpu readCpu(StreamInput in) throws IOException {
         @Override
         public void readFrom(StreamInput in) throws IOException {
             percent = in.readShort();
-            loadAverage = in.readDouble();
+            loadAverage = new LoadAverage();
+            loadAverage.readFrom(in);
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeShort(percent);
-            out.writeDouble(loadAverage);
+            loadAverage.writeTo(out);
         }
 
         public short getPercent() {
             return percent;
         }
 
-        public double getLoadAverage() {
+        public LoadAverage getLoadAverage() {
             return loadAverage;
         }
+
+        public static class LoadAverage implements Streamable {
+            private double oneMinuteLoadAverage;
+            private double fiveMinuteLoadAverage;
+            private double fifteenMinuteLoadAverage;
+
+            public final static LoadAverage DEFAULT_LOAD_AVERAGE = new LoadAverage(-1, -1, -1);
```

Comment:
@rmuir I removed `OsStats.Cpu.LoadAverage` and returned to using a `double[]` in 10863db685d52cd9d7e4c3217f739a1051d158f8. Personally I think that the dedicated class is simpler, but I have no issues keeping it this way if you feel otherwise.